### DISCUSSION TITLE
Show linkhints automatically

### DISCUSSION THREAD

**Discussion by ghost** [gdh1995/vimium-c#899] [Ideas]
I think it would be a great feature to have linkhints show all the time, but still only activate after pressing the LinkHints.activate keybind. Maybe they could be dimmed when not activate to show the difference. This would shave off the few seconds of inefficiency between the time when you click the LinkHints.activate key and then look to see what the letters are for the link you want. 

I know that linkhints refreshes the letters based on what links are in the browser viewpoint, so this would be tricky with the letters always changing while you scroll. Other than that, I think this feature wouldn't be too hard to implement.

Would anyone else be interested in this kind of thing? I might try to make it, although I'm really not good at browser stuff. Please let me know! :)

**Comment by gdh1995** [comment]
Um, I'm afraid the time and CPU cost is much more expensive than you may thought.


What's worse is, LinkHints' initing itself will work slower on bigger (and slower) pages, like a long message flow page.


I hate latency during my thoughts and real-world effects, too. Therefore, I also dislike that an extension I'm proud of will slow down users' workflow - this is why I haven't add this feature.





---Original---
From: ***@***.***&gt;
Date: Thu, Mar 23, 2023 06:14 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: [gdh1995/vimium-c] Show linkhints automatically (Discussion #899)




 
I think it would be a great feature to have linkhints show all the time, but still only activate after pressing the LinkHints.activate keybind. Maybe they could be dimmed when not activate to show the difference. This would shave off the few seconds of inefficiency between the time when you click the LinkHints.activate key and then look to see what the letters are for the link you want.
 
I know that linkhints refreshes the letters based on what links are in the browser viewpoint, so this would be tricky with the letters always changing while you scroll. Other than that, I think this feature wouldn't be too hard to implement.
 
Would anyone else be interested in this kind of thing? I might try to make it, although I'm really not good at browser stuff. Please let me know! :)
